eZ/Publish/Core/MVC/Symfony/Controller/Content/QueryController.php

```diff
@@ -81,4 +88,67 @@ private function runQuery(ContentView $view, $method)
         );
         $view->addParameters([$view->getParameter('query')['assign_results_to'] => $searchResults]);
     }
+
+    /**
+     * @param \eZ\Publish\Core\MVC\Symfony\View\ContentView $view
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @return \eZ\Publish\Core\MVC\Symfony\View\ContentView
+     */
+    public function pagingQueryAction(ContentView $view, Request $request)
+    {
+        $this->runPagingQuery($view, $request);
+
+        return $view;
+    }
+
+    /**
+     * @param \eZ\Publish\Core\MVC\Symfony\View\ContentView $view
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     */
+    private function runPagingQuery(ContentView $view, Request $request)
+    {
+        $queryParameters = $view->getParameter('query');
+
+        $limit = $this->getParameter('limit', 10, $queryParameters);
```

Comment:
How about simply `$queryParameters['limit'] ?? 10` instead of getting into design issues with `getParameter`?

Reply:
I think I'll take this idea and remove the getParameter method.